### PR TITLE
[TEST] Remove seqan3::test::tmp_filename in favor of tmp_directory.

### PR DESCRIPTION
**Pull request by smehringer**
Superseeds https://github.com/seqan/seqan3/pull/2753 because of staleness and because I had some time

### CODE REVIEW

File: test/unit/io/sequence_file/sequence_file_output_test.cpp [hunk 1]
```diff
@@ -412,11 +422,10 @@ TEST(columns, writing_id_seq_qual)
 // compression
 // ----------------------------------------------------------------------------
 
-std::string compression_by_filename_impl([[maybe_unused]] seqan3::test::tmp_filename & filename)
+std::string compression_by_filename_impl(seqan3::test::sandboxed_path const & filename)
 {
     {
-        seqan3::sequence_file_output fout{filename.get_path()};
-        fout.options.fasta_blank_before_id = true;
+        seqan3::sequence_file_output fout{filename};
```
Comment (eseiler) [inline test/unit/io/sequence_file/sequence_file_output_test.cpp:428]:
```suggestion
        seqan3::sequence_file_output fout{filename};
        fout.options.fasta_blank_before_id = true;
```

**Review by eseiler — APPROVED** [review]

**Comment by eseiler** [comment]
Which one first? This PR or https://github.com/seqan/seqan3/pull/3063 ?

**Comment by smehringer** [comment]
#3063 first because there are still some explicit usages of `clean()`

**Comment by eseiler** [comment]
Can be rebased

**Pull request merged**